### ISSUE TITLE
allowNonTsExtensions should also honour files without extensions

### ISSUE THREAD

**Issue by jrieken** [Microsoft/TypeScript#1593]
Set the ```allowNonTsExtensions``` to ```true``` and have a file without an extension. Pass it to the compiler and you will see a compiler error like ```cannot find file <name>.ts```. This is a problem when an editor has an unsaved buffer which mime type is known to be TypeScript.

**Comment by dsherret** [comment]
Was this meant to be included in 1.4?

http://stackoverflow.com/questions/33785337/tsc-unknown-compiler-option-allownontsextensions

**Comment by mhegazy** [comment]
@dsherret this is not a supported compiler flag. you can not pass this on the commandline. this flag is mainly for tooling purposes, e.g. Monaco web editor where file names do not have full extensions.

**Comment by mhegazy** [comment]
for future searchs, if you are using this flag to allow consumption of .js files, [`--allowJs`](https://github.com/Microsoft/TypeScript/pull/5471) is the right flag to use.
